app/presenters/publishing_api_presenters/unpublishing.rb

```diff
@@ -1,87 +1,76 @@
-class PublishingApiPresenters::Unpublishing
-  attr_reader :unpublishing, :update_type
+module PublishingApiPresenters
+  class Unpublishing < Item
+    def content
+      item.redirect? ? redirect_hash : super
+    end
 
-  def initialize(unpublishing, options = {})
-    @unpublishing = unpublishing
-    @update_type = options[:update_type] || default_update_type
-  end
+  private
+    def document_format
+      item.redirect? ? 'redirect' : 'unpublishing'
+    end
 
-  def as_json
-    if unpublishing.redirect?
-      redirect_hash
-    else
-      unpublishing_hash
+    def base_path
+      item.document_path
     end
-  end
 
-  def base_path
-    unpublishing.document_path
-  end
+    def redirect_hash
+      {
+        base_path: base_path,
+        format: 'redirect',
+        publishing_app: 'whitehall',
+        redirects: [
+          { path: base_path, type: 'exact', destination: alternative_path }
+        ],
+      }
+    end
 
-private
+    def edition
+      @edition ||= ::Edition.unscoped.find(item.edition_id)
+    end
 
-  def redirect_hash
-    {
-      content_id: unpublishing.content_id,
-      base_path: base_path,
-      format: 'redirect',
-      publishing_app: 'whitehall',
-      redirects: [
-        { path: base_path, type: 'exact', destination: alternative_path }
-      ],
-      update_type: update_type,
-    }
-  end
+    def rendering_app
+      edition.rendering_app
+    end
 
-  def unpublishing_hash
-    {
-      base_path: base_path,
-      content_id: unpublishing.content_id,
-      title: edition.title,
-      description: edition.summary,
-      format: 'unpublishing',
-      locale: I18n.locale.to_s,
-      need_ids: edition.need_ids,
-      public_updated_at: edition.public_timestamp,
-      update_type: update_type,
-      publishing_app: 'whitehall',
-      rendering_app: edition.rendering_app,
-      routes: [ { path: base_path, type: 'exact' } ],
-      redirects: [],
-      details: details,
-    }
-  end
+    def title
+      edition.title
+    end
 
-  def edition
-    @edition ||= Edition.unscoped.find(unpublishing.edition_id)
-  end
+    def description
+      edition.summary
+    end
 
-  def alternative_path
-    full_uri = Addressable::URI.parse(unpublishing.alternative_url)
+    def public_updated_at
+      edition.public_timestamp
+    end
 
-    path_uri = Addressable::URI.new
-    path_uri.path = full_uri.path
-    path_uri.query = full_uri.query
-    path_uri.fragment = full_uri.fragment
+    def need_ids
+      edition.need_ids
+    end
 
-    path_uri.to_s
-  end
+    def alternative_path
+      full_uri = Addressable::URI.parse(item.alternative_url)
 
-  def details
-    {
-      explanation: unpublishing_explanation,
-      unpublished_at: unpublishing.created_at,
-      alternative_url: unpublishing.alternative_url
-    }
-  end
+      path_uri = Addressable::URI.new
```

Comment:
`new` takes an options hash including `path`, `query`, `fragment` and friends. That might tidy this up a teeny bit.